src/CapabilityService.Tests/Scenarios/DeleteCapabilityScenario.cs

```diff
@@ -0,0 +1,77 @@
+using System;
```

Comment:
I have introduced the Gherkin syntax to our code base in attempt express expectations to interactions between classes. I feel it can be difficult to express derived effects of actions in the scope of a classical unit test. Hence Gherkin and a attempt at BDD.
We did some quick tests of 7ish different frameworks when we wrote the first scenario and my conclusion so far is that all of them had many opinions about how to run a scenario, many felt cumbersome and none felt awesome to use.
I asked myself what what do I want from a scenario test, and the top 2 things are
- Tests that covers interactions between classes.
- A easy to read description of steps in the test.

I feel the current solution has atomblished this very well.
That said we can investigate if we can elevate our work even more by applying a framework.

Reply:
We worked with https://specflow.org/ (Cucumber) in on of my old projects, which was quiet cool. However my main concerns with the current approach are two fold:

1) Is that we will end up doing "test-last", rather then "test-first"?

2) We will not be able to train business analysts to write "specifications" for us, which means we cant offload the BDD work to the people who design the actual behavior we encode in our software, which is sub-optimal and defeats the other half of why BDD makes sense.